unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/AvatarSlot/AvatarSlotsHUDController.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace DCL.Backpack
+{
+    public class AvatarSlotsHUDController : IHUD
+    {
+        internal AvatarSlotsDefinitionSO avatarSlotsDefinition;
+        internal readonly IAvatarSlotsView avatarSlotsView;
+        internal string lastSelectedSlot;
+
+        public AvatarSlotsHUDController(IAvatarSlotsView avatarSlotsView)
+        {
+            this.avatarSlotsView = avatarSlotsView;
+            avatarSlotsView.OnToggleAvatarSlot += ToggleSlot;
+            avatarSlotsDefinition = Resources.Load<AvatarSlotsDefinitionSO>("AvatarSlotsDefinition");
```

Comment:
As we were discussing in our last meeting with @lorux0, you and me, this specific behavior of getting the slots configuration from a SO should be abstracted into an interface, for example `IAvatarSlotsConfigProvider`, and have a specific implementation for getting the config from a SO (in the future we could have another specific implementation that would use the same interfaz but for getting the config from backend for example). The idea would be to inject that provider interface in this controller and abstract it from the implementation (it would be basically to follow a Strategy pattern).

Reply:
Yes I understand and it makes sense but I would do that once we go into the next phase of the avatar v2, if ever

Reply:
Ok, understood... although I think it would be a good practice to abstract the logic. It would be an easy change. Anyway, I agree with this option if we all decide to follow it 👍 Wdyt @lorux0?